release.py

```diff
@@ -0,0 +1,498 @@
+            version_str = match.group(1)
+        else:
+            content = file_path.read_text()
+            match = re.search(PYPROJECT_VERSION_STR, content)
+            if not match:
+                raise ValueError(f'Could not find version string in {file}')
+            version_str = match.group(1)
+
+        current_version, dev_suffix = parse_version(version_str)
+        if dev_suffix:
+            raise ValueError(f'Version already has dev suffix: {current_version}')
+
+        new_version = bump_version(current_version) + '.dev0'
+        if 'ops' in file:
+            update_ops_version(file, new_version)
+        else:
+            update_pyproject_version(file, new_version)
+        logger.info(f'Updated {file} to post-release version: {new_version}')
+
+
+def countdown(msg: str, t: int):
+    """Countdown timer to show before exiting."""
+    while t:
+        timer = f'00:{t:02d}'
+        print(f'{msg}: {timer}', end='\r')
+        time.sleep(1)
+        t -= 1
+
+
+def check_update_charm_pins_prs(repo: github.Repository.Repository):
+    """Check for open PRs that update charm pins."""
+    prs = repo.get_pulls(state='open')
+    open_prs = [pr for pr in prs if pr.title == 'chore: update charm pins']
+    if open_prs:
+        logger.info('Please merge "update charm pins" PRs first:')
+        pr = open_prs[0]
+        pr_url = f'#{pr.number} - {pr.html_url}'
+        logger.info(f'  {pr_url}')
+        if len(open_prs) > 1:
+            logger.info(
+                f'Note that there are {len(open_prs) - 1} more open update charm pins PRs.'
+            )
+        countdown('Exiting and opening the charm pins PR in: ', 3)
+        webbrowser.open(pr.html_url)
+        exit(1)
+
+
+def release(owner: str, repo_name: str, branch: str):
+    """Create a release, update changelog, and create a PR for the release."""
+    org = gh_client.get_organization(owner)
+    repo = org.get_repo(repo_name)
+
+    check_update_charm_pins_prs(repo)
+
+    tag = get_new_version(owner, repo, branch)
+
+    release = create_draft_release(repo, tag, branch)
+    if not release:
+        logger.error('Failed to create draft release.')
+        exit(1)
+    logger.info(f'Draft release created: {release.html_url}.')
+
+    categories, full_changelog = parse_release_notes(release.body)
+    notes = format_notes(categories, full_changelog)
+    print_notes(notes)
+    title, summary = get_title_and_summary(release.title)
+    if not title:
+        title = tag
+    notes = f'{summary}\n{notes}'
+    update_release(release, title, notes)
+
+    changes = format_changes(categories, tag)
+    update_changes_file(changes, 'CHANGES.md')
+
+    update_versions_for_release(tag)
+
+    new_branch = f'release-prep-{tag}'
+    subprocess.run(['/usr/bin/git', 'checkout', '-b', new_branch], check=True)
+    changed_files = ['CHANGES.md', *VERSION_FILES]
+    for file in changed_files:
+        subprocess.run(['/usr/bin/git', 'add', file], check=True)
```

Comment:
I think that we need to validate that each file was in fact changed.
`git add` returns OK status even if there are no changes.

Perhaps it's better to fail early in the `update_versions_**` or `update_ops/pyproject_version` functions.